compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -820,6 +805,25 @@ public void visit(BLangPackage pkgNode) {
         result = pkgNode;
     }
 
+    private void constantRewrite(BLangPackage pkgNode, BLangBlockFunctionBody initFnBody) {
+        for (BLangConstant constant : pkgNode.constants) {
+            BType constType = types.getReferredType(constant.symbol.type);
+            if (constType.tag == TypeTags.INTERSECTION) {
```

Comment:
You can check the inverse and continue here. That way we can get rid of the indentation in L812-L822.